### DISCUSSION TITLE
Observation - Enabling just the Gyro Sensor, On the callback event, I get back all 3 sensors in succession: Accelerometer, Gyroscope, Magnetometer

### DISCUSSION THREAD

**Discussion by ricksticks1** [everysight-maverick/sdk#5] [General]
Hi,
Love the inertial sensors in the Maverick, but noticed when I enabled only 1 sensor (Gyroscope), on the callback event, I get back all 3 sensors in succession: Accelerometer, Gyroscope, Magnetometer. Just wanted to verify if I am doing this correctly :)

```
**Example enable sensor:**   
          
 private fun enableGyroSensor() {
                          with(Evs.instance().sensors() ){
                          enableGyro(SensorRate.rate1)
                          enableSensorsFusion(SensorRate.rate1, false)
                      }
                  }
```
 ```
**Example sensors callback event:**

private val gyroEvent = object : IEvsInertialSensorsEvents {
                     override fun onSensor(
                sensorType: SensorType,
                timestampNs: Long,
                x: Float,
                y: Float,
                z: Float
            ) {
                when (sensorType) {
                    SensorType.acc -> angleText = Log.d(TAG, "accl data recv")
                    SensorType.gyro -> compass.setWorldInertial(x, y, z)
                    SensorType.mag -> angleText = Log.d(TAG, "magnetic data recv")
                    else -> { Log.d(TAG, "** other sensor **") }
                }
            }
        }
```

Thanks,
Rick

Rick Strickland 
USA - North Carolina

**Comment by feranf** [comment]
Hi Rick,

There are two ways to consume Maverick sensors:

1. [Base Sensors](https://everysight.github.io/maverick_docs/sdk-engine/sensors/#base-sensors) ,which are factory calibrated and can be consumed separately.
2. [Fused Sensors](https://everysight.github.io/maverick_docs/sdk-engine/sensors/#fused-composite) , which are activated via `enableSensorsFusion`

The `enableSensorsFusion` method automatically activates all the sensors - which is why you receive data from all of them. If you only want to receive data from the Gyro, do not call `enableSensorsFusion`; instead, call `enableGyro`

If you wish to use sensor fusion, call `enableSensorsFusion` ( and as a result the `enableGyro` will be called internally and the fusion logic will be activated)

Hope that helped,

**Comment by ricksticks1** [comment]
Hi,
Thank you, yes I did try with just `enableGyro` without the `enableSensorsFusion`, but it will not callback any sensor data (accl, gyro, magnetic).

This works (unfortunately multiple sensors on callback `onSensor`):
```
private fun enableGyroSensor() {
        Evs.instance().sensors().enableGyro(SensorRate.rate1)
        Evs.instance().sensors().enableSensorsFusion(SensorRate.rate1, false)
}
```

Not working, no data callback `onSensor` :
```
private fun enableGyroSensor() {
        Evs.instance().sensors().enableGyro(SensorRate.rate1)
   
}
```

According to the documentation, it is a required step to have `enableSensorsFusion` to have callback data for `onSensor` event:

**IEvsInertialSensorsEvents**    https://everysight.github.io/maverick_docs/api-reference/-everysight%20-s-d-k/UIKit.services/-i-evs-inertial-sensors-events/index.html

**interface IEvsInertialSensorsEvents**
Interface for receiving calibrated sensors data (Accelerometer, Magnetometer, Gyroscope)

To receive the sensors, for supported systems, you should:

1. Enable the sensor via IEvsSensorsService.enableAccelerometer, IEvsSensorsService.enableMagnetometer, IEvsSensorsService.enableGyro
2. Enable sensor fusion IEvsSensorsService.enableSensorsFusion
3. Register the sensor data callback via IEvsSensorsService.registerInertialSensorsEvents


Thank you,
Rick

**Reply by feranf** [reply]
Hi Rick,

Thank you for the detailed information.
I'll open an issue so the development team will look into it.

Meanwhile, I suggest to continue with the `enableSensorsFusion` and using only the Gyro from the received events.

Eran
